jobsdb/jobsdb.go

```diff
@@ -1087,7 +1092,7 @@ func (jd *HandleT) refreshDSRangeList(l lock.DSListLockToken) {
 			continue
 		}
 
-		if idx < len(dsList)-1 && (jd.inProgressMigrationTargetDS == nil || jd.inProgressMigrationTargetDS.Index != ds.Index) {
```

Comment:
because we are now performing the whole migration in a single transaction. So `refreshDSRangeList` no longer has to worry about in-progress datasets since it cannot see them until they actually become part of the dataset.